components/server-side-render/index.js

```diff
@@ -20,46 +27,56 @@ export class ServerSideRender extends Component {
 		};
 	}
 
+	componentWillMount() {
+		this.isStillMounted = true;
+	}
+
 	componentDidMount() {
 		this.fetch( this.props );
 	}
 
+	componentWillUnmount() {
+		this.isStillMounted = false;
+	}
+
 	componentWillReceiveProps( nextProps ) {
 		if ( ! isEqual( nextProps, this.props ) ) {
 			this.fetch( nextProps );
 		}
 	}
 
 	fetch( props ) {
-		this.setState( { response: null } );
+		if ( null !== this.state.response ) {
+			this.setState( { response: null } );
+		}
 		const { block, attributes } = props;
 
 		const path = '/gutenberg/v1/block-renderer/' + block + '?context=edit&' + this.getQueryUrlFromObject( { attributes } );
 
-		return wp.apiRequest( { path: path } ).then( ( response ) => {
-			if ( response && response.rendered ) {
+		return apiRequest( { path: path } ).then( ( response ) => {
+			if ( this.isStillMounted && response && response.rendered ) {
 				this.setState( { response: response.rendered } );
 			}
 		} );
 	}
 
 	getQueryUrlFromObject( obj, prefix ) {
 		return map( obj, ( paramValue, paramName ) => {
-			const key = prefix ? prefix + '[' + paramName + ']' : paramName,
-				value = obj[ paramName ];
-			return isObject( paramValue ) ? this.getQueryUrlFromObject( value, key ) :
-				encodeURIComponent( key ) + '=' + encodeURIComponent( value );
+			const key = prefix ? prefix + '[' + paramName + ']' : paramName;
+			return isPlainObject( paramValue ) ? this.getQueryUrlFromObject( paramValue, key ) :
+				encodeURIComponent( key ) + '=' + encodeURIComponent( paramValue );
 		} ).join( '&' );
 	}
 
 	render() {
 		const response = this.state.response;
-		if ( ! response || ! response.length ) {
+		if ( ! response ) {
```

Comment:
Unrelated to this PR, but should we handle a possible error state here? If the API returns with a 404 or 500 response then we'll be stuck showing a spinner forever.

Reply:
All the comments should be addressed within 9df912a, also added a possible handling for error response including displaying the error message from REST API in case of failure.
For example:
![screen shot 2018-05-10 at 4 47 46 pm](https://user-images.githubusercontent.com/3294597/39874756-68b2238e-5477-11e8-8c89-90b9967d2b3c.png)

Let me know if this approach seems OK.

Reply:
Yeah this works for me. It matches the error state that Shared Blocks have which is nice. 

Going forward we probably will want to improve the error state for all of our dynamic blocks. For example, we should make sure that they all have actionable error messages and e.g. _Try again_ buttons. 

But I don't want that to block this PR so 👍 for now.